src/EoS/Nuclear/CPU_EoS_Nuclear.cpp

```diff
@@ -454,18 +481,266 @@ static real EoS_DensPres2CSqr_Nuclear( const real Dens_Code, const real Pres_Cod
+GPU_DEVICE_NOINLINE
+static void EoS_General_Nuclear( const int Mode, real Out[], const real In[], const double AuxArray_Flt[],
+                                 const int AuxArray_Int[], const real *const Table[EOS_NTABLE_MAX] )
+{
+
+// general check
+#  ifdef GAMER_DEBUG
+   if ( Out          == NULL )   printf( "ERROR : Out == NULL in %s !!\n", __FUNCTION__ );
+   if ( In           == NULL )   printf( "ERROR : In == NULL in %s !!\n", __FUNCTION__ );
+   if ( AuxArray_Flt == NULL )   printf( "ERROR : AuxArray_Flt == NULL in %s !!\n", __FUNCTION__ );
+   if ( AuxArray_Int == NULL )   printf( "ERROR : AuxArray_Int == NULL in %s !!\n", __FUNCTION__ );
+#  endif // GAMER_DEBUG
+
+
+   switch ( Mode )
+   {
+//    temperature mode
+      case NUC_MODE_TEMP :
+      {
+         const real Dens_Code = In[0];
+         const real Temp_Kelv = In[1];
+         const real Ye        = In[2];
+
+//       check
+#        ifdef GAMER_DEBUG
+         if ( Hydro_CheckNegative(Dens_Code) )
+            printf( "ERROR : invalid input density (%14.7e) at file <%s>, line <%d>, function <%s>\n",
+                    Dens_Code, __FILE__, __LINE__, __FUNCTION__ );
+
+         if ( Hydro_CheckNegative(Temp_Kelv) )
+            printf( "ERROR : invalid input temperature (%14.7e) at file <%s>, line <%d>, function <%s>\n",
+                    Temp_Kelv, __FILE__, __LINE__, __FUNCTION__ );
+
+         if ( Hydro_CheckNegative(Ye) )
+            printf( "ERROR : invalid input Ye (%14.7e) at file <%s>, line <%d>, function <%s>\n",
+                    Ye, __FILE__, __LINE__, __FUNCTION__ );
+#        endif // GAMER_DEBUG
+
+         const real EnergyShift = AuxArray_Flt[NUC_AUX_ESHIFT    ];
+         const real Dens2CGS    = AuxArray_Flt[NUC_AUX_DENS2CGS  ];
+         const real Kelvin2MeV  = AuxArray_Flt[NUC_AUX_KELVIN2MEV];
+         const real sEint2Code  = AuxArray_Flt[NUC_AUX_VSQR2CODE ];
+
+         const int  NRho        = AuxArray_Int[NUC_AUX_NRHO ];
+         const int  NEps        = AuxArray_Int[NUC_AUX_NEPS ];
+         const int  NYe         = AuxArray_Int[NUC_AUX_NYE  ];
+         const int  NMode       = AuxArray_Int[NUC_AUX_NMODE];
+
+         real Dens_CGS  = Dens_Code * Dens2CGS;
+         real Temp_MeV  = Temp_Kelv * Kelvin2MeV;
+         real sEint_CGS = NULL_REAL;
+         real Entr      = NULL_REAL;
+         real Useless   = NULL_REAL;
+         int  Err       = NULL_INT;
+
+
+//       check floating-point overflow and Ye
+#        ifdef GAMER_DEBUG
+         if ( Nuc_Overflow(Dens_CGS) )
+            printf( "ERROR : EoS overflow (Dens_CGS %13.7e, Dens_Code %13.7e, Dens2CGS %13.7e) in %s() !!\n",
+                    Dens_CGS, Dens_Code, Dens2CGS, __FUNCTION__ );
+
+         if ( Nuc_Overflow(Temp_MeV) )
+            printf( "ERROR : EoS overflow (Temp_MeV %13.7e, Temp_Kelv %13.7e, Kelvin2MeV %13.7e) in %s() !!\n",
+                    Temp_MeV, Temp_Kelv, Kelvin2MeV, __FUNCTION__ );
+
```

Comment:
Since no line number output here, it could be better to output which quantity is overflow and which mode is used? Same for the entropy mode.

Reply:
I just realized the name of quantity is already output here. So maybe add which mode is used?